Search/QuickSelect.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * General Selection: Finding the kth smallest number
+ *
+ * Notes: QuickSelect is related to QuickSort, thus has optimal best and average
+ * case (O(n)) but poor worst case (O(n^2))
+ *
+ * @flow
+ */
+
+import { expect } from 'chai'
+
+
+type num = number
+
+export default function QuickSelect(items: num[], i: num): num {
+  const itemCopy = [...items]
+  return RandomizedSelect(itemCopy, 0, items.length - 1, i)
+}
+
+function RandomizedSelect(items: num[], p: num, r: num, i: num): any {
```

Comment:
Would be great if you could convert the one char variable names to short phrases. Or you can create a docblock with the types.

ex

```js
/**
  * @param  p paramater: Some description of the param here
  */
```